components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -132,6 +139,29 @@ type ideRoutes struct {
 	workspaceMustExistHandler mux.MiddlewareFunc
 }
 
+func (ir *ideRoutes) HandleSSHHostKeyRoute(route *mux.Route, hostKeyList []ssh.Signer) {
+	shk := make([]struct {
+		Type    string `json:"type"`
+		HostKey string `json:"host_key"`
+	}, len(hostKeyList))
+	for i, hk := range hostKeyList {
+		shk[i].Type = hk.PublicKey().Type()
+		shk[i].HostKey = base64.StdEncoding.EncodeToString(hk.PublicKey().Marshal())
+	}
+	byt, err := json.Marshal(shk)
+	if err != nil {
+		log.WithError(err).Error("ssh_host_key router setup failed")
```

Comment:
this would produce a subtle failure mode: we'd get this error message once, but SSH access would be broken.

Not a reason not to merge, but maybe a follow up. Instead of just logging, we could bubble up the error and exit.